### PR TITLE
Add a --replay option to ::execute to avoid mongodb unique index error

### PR DESCRIPTION
**Pull request by redthor**
For #11

### CODE REVIEW

File: tests/AntiMattr/Tests/MongoDB/Migrations/Tools/Console/Command/ExecuteCommandTest.php [hunk 1]
```diff
@@ -117,7 +117,52 @@ public function testExecuteUpWithInteraction()
...
+        ;
+
+        // Run command, run.
+        $this->command->run(
+            $input,
+            $this->output
+        );
+    }
+
+    public function testExecuteReplyWithoutInteraction()
```
Comment (trq) [inline tests/AntiMattr/Tests/MongoDB/Migrations/Tools/Console/Command/ExecuteCommandTest.php]: Should be `testExecuteReplayWithoutInteraction`

Comment (redthor) [inline tests/AntiMattr/Tests/MongoDB/Migrations/Tools/Console/Command/ExecuteCommandTest.php]: good one, thanks

File: src/AntiMattr/MongoDB/Migrations/Version.php [hunk 2]
```diff
@@ -276,13 +284,24 @@ public function executeScript(Database $db, $file)
...
     {
         $this->configuration->createMigrationCollection();
         $collection = $this->configuration->getCollection();
 
         $document = array('v' => $this->version, 't' => $this->createMongoTimestamp());
-        $collection->insert($document);
+
+        if ($replay) {
+            $query = array('v' => $this->version);
+            $collection->update($query, $document);
```
Comment (caciobanu) [inline src/AntiMattr/MongoDB/Migrations/Version.php]: What happens when we use "replay" and there was no previous execution of the this version ?

Comment (caciobanu) [inline src/AntiMattr/MongoDB/Migrations/Version.php]: Maybe the following is better ?
```php
$collection->update($query, $document, array('upsert' => true));
```

Comment (redthor) [inline src/AntiMattr/MongoDB/Migrations/Version.php]: Thank you for this @caciobanu ! I had intended for an upsert as originally discussed and missed it!

**Comment by caciobanu** [comment]
LGTM

**Pull request merged**